pkg/controllers/verticadb_controller.go

```diff
@@ -149,14 +150,19 @@ func (r *VerticaDBReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		MakeRebalanceShardsReconciler(r, log, vdb, prunner, &pfacts),
 	}
 
+	// Iterate over each actor
 	for _, act := range actors {
 		log.Info("starting actor", "name", fmt.Sprintf("%T", act))
 		res, err = act.Reconcile(ctx, &req)
 		// Error or a request to requeue will stop the reconciliation.
-		if err != nil || res.Requeue {
-			if res.Requeue && vdb.Spec.RequeueTime > 0 {
+		if verrors.IsReconcileAborted(res, err) {
+			// Handle requeue time priority.
+			// If any function needs a requeue and we have a RequeueTime set,
+			// then overwrite RequeueAfter.
+			// Functions such as Upgrade may already set RequeueAfter and Requeue to false
+			if res.RequeueAfter > 0 && vdb.Spec.RequeueTime > 0 {
```

Comment:
I think the check should be:
`if (res.Requeue || res.RequeueAfter > 0) && vdb.Spec.RequeueTime > 0`

That way any place that previously set Requeue to true will be overridden.

Reply:
Good point. Thanks for the comments. Will add the changes